test/system/src/test/java/io/pravega/test/system/MultiReaderWriterTxnWithFailoverTest.java

```diff
@@ -0,0 +1,444 @@
+        eventWriteCount = new AtomicLong(0);
+
+        //create a scope
+        try (StreamManager streamManager = new StreamManagerImpl(controllerUri)) {
+            Boolean createScopeStatus = streamManager.createScope(scope);
+            log.info("Creating scope with scope name {}", scope);
+            log.debug("Create scope status {}", createScopeStatus);
+            //create a stream
+            Boolean createStreamStatus = streamManager.createStream(scope, STREAM_NAME, config);
+            log.debug("Create stream status {}", createStreamStatus);
+        }
+
+        //get ClientFactory instance
+        log.info("Scope passed to client factory {}", scope);
+        try (ClientFactory clientFactory = new ClientFactoryImpl(scope, controller);
+             ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerUri)) {
+
+            log.info("Client factory details {}", clientFactory.toString());
+            //create writers
+            log.info("Creating {} writers", NUM_WRITERS);
+            List<EventStreamWriter<Long>> writerList = new ArrayList<>();
+            log.info("Writers writing in the scope {}", scope);
+            for (int i = 0; i < NUM_WRITERS; i++) {
+                log.info("Starting writer{}", i);
+                final EventStreamWriter<Long> writer = clientFactory.createEventWriter(STREAM_NAME,
+                        new JavaSerializer<Long>(),
+                        EventWriterConfig.builder().retryAttempts(10).build());
+                writerList.add(writer);
+            }
+
+            //create a reader group
+            log.info("Creating Reader group : {}", readerGroupName);
+            log.info("Scope passed to readergroup manager {}", scope);
+
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder().startingTime(0).build(),
+                    Collections.singleton(STREAM_NAME));
+            log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+            log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+            log.info("Online readers {}", readerGroupManager.getReaderGroup(readerGroupName).getOnlineReaders());
+
+            //create readers
+            log.info("Creating {} readers", NUM_READERS);
+            List<EventStreamReader<Long>> readerList = new ArrayList<>();
+            String readerName = "reader" + new Random().nextInt(Integer.MAX_VALUE);
+            log.info("Scope that is seen by readers {}", scope);
+            //start reading events
+            for (int i = 0; i < NUM_READERS; i++) {
+                log.info("Starting reader{}", i);
+                log.info("Creating reader with id {}", readerName + i);
+                final EventStreamReader<Long> reader = clientFactory.createReader(readerName + i,
```

Comment:
The readerName has both a random integer and a sequential one appended to it. This seems redundant. Either would be sufficient to ensure uniqueness.